addons/sale_expense/tests/test_reinvoice.py

```diff
@@ -8,243 +8,381 @@
+            # [0] Line not created from a re-invoiced, should never be changed
             {'qty_delivered': 0.0, 'product_uom_qty': 3.0, 'is_expense': False, 'name': 'expense_employee: expense_1 invoicing=order, expense=sales_price'},
+            # [1-6] SHEET 1 Lines: created with the correct quantities and linked to the expense
+            {'qty_delivered': 6.0, 'product_uom_qty': 6.0, 'is_expense':  True, 'name': 'expense_employee: expense_6 invoicing=order, expense=cost'},
+            {'qty_delivered': 5.0, 'product_uom_qty': 5.0, 'is_expense':  True, 'name': 'expense_employee: expense_5 invoicing=delivery, expense=cost'},
+            {'qty_delivered': 1.0, 'product_uom_qty': 1.0, 'is_expense':  True, 'name': 'expense_employee: expense_4 invoicing=delivery, expense=sales_price'},
+            {'qty_delivered': 1.0, 'product_uom_qty': 1.0, 'is_expense':  True, 'name': 'expense_employee: expense_3 invoicing=delivery, expense=sales_price'},
+            {'qty_delivered': 1.0, 'product_uom_qty': 1.0, 'is_expense':  True, 'name': 'expense_employee: expense_2 invoicing=order, expense=sales_price'},
             {'qty_delivered': 1.0, 'product_uom_qty': 1.0, 'is_expense':  True, 'name': 'expense_employee: expense_1 invoicing=order, expense=sales_price'},
-            {'qty_delivered': 2.0, 'product_uom_qty': 2.0, 'is_expense':  True, 'name': 'expense_employee: expense_2 invoicing=order, expense=sales_price'},
-            {'qty_delivered': 3.0, 'product_uom_qty': 3.0, 'is_expense':  True, 'name': 'expense_employee: expense_3 invoicing=delivery, expense=sales_price'},
-            {'qty_delivered': 4.0, 'product_uom_qty': 4.0, 'is_expense':  True, 'name': 'expense_employee: expense_4 invoicing=delivery, expense=sales_price'},
-            {'qty_delivered': 3.0, 'product_uom_qty': 3.0, 'is_expense':  True, 'name': 'expense_employee: expense_3 invoicing=delivery, expense=sales_price copy'},
-            {'qty_delivered': 4.0, 'product_uom_qty': 4.0, 'is_expense':  True, 'name': 'expense_employee: expense_4 invoicing=delivery, expense=sales_price copy'},
+            # [7-8] SHEET 2 Lines: should never be changed
+            {'qty_delivered': 1.0, 'product_uom_qty': 1.0, 'is_expense':  True, 'name': 'expense_employee: expense_4 invoicing=delivery, expense=sales_price copy'},
+            {'qty_delivered': 1.0, 'product_uom_qty': 1.0, 'is_expense':  True, 'name': 'expense_employee: expense_3 invoicing=delivery, expense=sales_price copy'},
         ])
 
-        expense_sheet.action_unpost()  # Lines [0-4] quantities are set to 0
-        self.assertRecordValues(sale_order.order_line, [
+    def test_expenses_reinvoice_case_2_reset_sheet_to_draft(self):
+        """
+        CASE 2: Reset to draft of the (first) expense sheet, the quantities of the corresponding SOL are set to 0
+        """
+        # CASE 1 steps
+        self.sale_expense_both_sheets._do_approve()
+        self.sale_expense_both_sheets.action_sheet_move_create()
```

Comment:
It looks like you are always doing this?

Reply:
Because of test_expenses_reinvoice_case_1_create_moves, as I don't want to assert in the setup and I prefer having the triggering action inside the test (did move the first _do_approve in the setup, as it doesn't make sense doing it multiple times)